### PR TITLE
feat(AUTH-CPC-310): Register user

### PR DESCRIPTION
**Pull request by Fube**
### Changes

- Added encoding logic (using Bcrypt)
- Added spring security rule to allow access to registration endpoint with role or authentication

### CODE REVIEW

File: auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java [hunk 1]
```diff
@@ -35,15 +36,10 @@ public User getUserById(long id) {
...
 
-        try {
-            log.info("Saving user with email {}", userIDLessDTO.getEmail());
-            User user = userMapper.idLessDTOToModel(userIDLessDTO);
-            return userRepo.save(user);
-        }
-        catch (DataIntegrityViolationException e){
-            throw new DataIntegrityViolationException("Duplicate email for userEmail " + userIDLessDTO.getEmail());
-        }
-
```
Comment (JordanAlbayrak) [inline auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java:46]: I believe this was added by Jeremy for his email validation/verification, I'm wondering if maybe this should have stayed. Next class I'll bring this up to confirm.

**Review by JordanAlbayrak — APPROVED** [review]
Thank you for taking the to complete this. It looks good to me, I left a comment not sure if it will show up but I'll ask you in class later anyway.

File: auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java (hunk 1 above)
Comment (Fube) [reply auth-service/src/main/java/com/petclinic/auth/User/UserServiceImpl.java:46]: The issue with this is that it assumes that ALL data integrity violations are due to duplicate emails.
This is untrue as a data integrity exception can occur when referencing a non-existent field or using an invalid value (in terms of type) and a myriad of other things.

**Review by Trilikin21 — APPROVED** [review]
LGTM

**Pull request merged**